nflog/nflog_test.go

```diff
@@ -157,15 +157,14 @@ func TestWithMaintenance_SupportsCustomCallback(t *testing.T) {
 			return 0, nil
 		})
 	}()
-	gosched()
 
 	// Before the first tick, no maintenance executed.
-	clock.Add(99 * time.Millisecond)
+	clock.Advance(99 * time.Millisecond)
 	require.EqualValues(t, 0, calls.Load())
 
 	// Tick once.
-	clock.Add(1 * time.Millisecond)
-	require.EqualValues(t, 1, calls.Load())
+	clock.Advance(1 * time.Millisecond)
+	require.Eventually(t, func() bool { return calls.Load() == 1 }, 5*time.Second, time.Second)
```

Comment:
This is now required as otherwise the `require.EqualValues` is run before the goroutine. This didn't happen with `benbjohnson/clock` as it included additional synchronization code to ensure all timers fired before `clock.Add` returned. This made tests easier to write, but is not how the `time` package behaves in real life.